core/adapters/eth_tx.go

```diff
@@ -260,13 +292,22 @@ func getTxDataUsingABIEncoding(e *EthTx, inputData models.JSON) ([]byte, error)
 				var arg [32]byte
 				copy(arg[:], b)
 				values[i] = arg
+			} else if argType == "address" {
```

Comment:
I intended to re-use this getTxDataUsingABIEncoding for the second level encoding (for the operator contract), so I added support for the bytes4 and address. Then it seemed like more trouble than its worth because I'd have to parse out all the preceding args from dataPrefix just to re-encode them again. Still, supporting more types is useful so I've kept them in here.

Reply:
What do you mean by "second level encoding"?

Reply:
dataPrefix seems like such a hack. Is there any way we can get away from it in the runnerv2 implementation of eth_tx?

Reply:
I'd much rather see either:

"well defined standard argument encoding with function selector"

OR

"raw binary passthrough"

Reply:
I'm thinking of the calldata argument to fulfillOracleRequest2 as the first argument encoding (calldata for the client's callback fn) and the second being the argument encoding of the fulfillOracleRequest2

Reply:
The stuff is fairly well defined (here)[https://github.com/smartcontractkit/chainlink/blob/08caa9f7de4ce0cd251d6e2c2793ec56fed452d3/core/store/models/log_events.go#L44]. I'm not sure of a way to get away from it, its actually a simple way to just re-use the binary log data as calldata. Alternatively you could parse the fields from the log and re-encoding them for the fulfill call for clarity, I'm not totally opposed to that just for readability